### ISSUE TITLE
[mqtt.homie] Missing support for "Duration" and "DateTime"

### ISSUE THREAD

**Issue by ccoors** [openhab/openhab-addons#10839]
## Expected Behavior
`org.openhab.binding.mqtt.homie` should support [Duration](https://homieiot.github.io/specification/spec-core-v3_0_0/#duration) and [DateTime](https://homieiot.github.io/specification/spec-core-v3_0_0/#datetime) as specified in the Homie 3.0.0 specification.

## Current Behavior
Adding a Homie-Thing in openHAB leads to multiple exceptions looking like this on the console:

```
java.lang.IllegalArgumentException: No enum constant org.openhab.binding.mqtt.homie.internal.homie300.PropertyAttributes.DataTypeEnum.duration_
        at java.base/java.lang.Enum.valueOf(Enum.java:240)
        at org.openhab.binding.mqtt.generic.mapping.SubscribeFieldToMQTTtopic.numberConvert(SubscribeFieldToMQTTtopic.java:114)
        at org.openhab.binding.mqtt.generic.mapping.SubscribeFieldToMQTTtopic.processMessage(SubscribeFieldToMQTTtopic.java:152)
        at org.openhab.core.io.transport.mqtt.internal.Subscription.processMessage(Subscription.java:85)
```
<details>
<summary>Click to see more of the exception...</summary>

```
        at org.openhab.core.io.transport.mqtt.internal.Subscription.lambda$1(Subscription.java:81)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.concurrent.ConcurrentHashMap$KeySpliterator.forEachRemaining(ConcurrentHashMap.java:3566)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:661)
        at org.openhab.core.io.transport.mqtt.internal.Subscription.messageArrived(Subscription.java:81)
        at org.openhab.core.io.transport.mqtt.internal.Subscription.messageArrived(Subscription.java:66)
        at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3AsyncClientView.lambda$callbackView$1(Mqtt3AsyncClientView.java:76)
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:303)
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:288)
        at com.hivemq.client.rx.FlowableWithSingle$SingleFutureSubscriber.onNext(FlowableWithSingle.java:406)
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber$Default.tryOnNextActual(FlowableWithSingleCombine.java:235)
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber.tryOnNext(FlowableWithSingleCombine.java:200)
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:649)
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```
</details>

It then brings the thing into the `COMMUNICATION_ERROR` state.

![image](https://user-images.githubusercontent.com/8963459/121415852-6f898f80-c968-11eb-9679-997e0cb9103c.png)

## Possible Solution
Probably map DateTime to the DateTime item type of openHAB and Duration to Number, as there doesn't seem to be a "Duration" type (according to [this](https://www.openhab.org/docs/concepts/items.html)).

## Context

I'm trying to add my vacuum cleaner running [Valetudo](https://valetudo.cloud) to openHAB. It publishes the state of various consumables (brush, filter, sensors...) to MQTT according to the Homie specification with type `Duration`.

![image](https://user-images.githubusercontent.com/8963459/121660403-84b70900-caa3-11eb-9108-cac8fd8095a9.png)

## Your Environment

* Version used: openHAB 3.1.0.M5 (Docker), MQTT Binding 3.1.0.M5
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...): Firefox 89
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): Desktop, Arch Linux x86_64

**Comment by Flunkyball** [comment]
Any idea when the proposed solution will be integrated ?
I am currently on OH 3.2 M3 and the problem still remains.

**Comment by ingoschi** [comment]
I just readded my Dreame D9 with 2021.11.0 with Homie enabled to openHAB 3.1.0.
First I also got no channels. But a few minutes later, after the CompletionException occured in the logs,
I could enable the thing and got all channels.
I remember I had the same starting problems the first time I added the robot to openHAB.
So great that now fix  #1200 for the error exists but the error did not prevent adding Dreamies forever.

openhab.log:
2021-11-07 19:50:59.917 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'mqtt:homie300:e0c52ded0e:dreameD9' to inbox.
2021-11-07 19:51:34.949 [WARN ] [core.thing.internal.ThingManagerImpl] - Initializing handler for thing 'mqtt:homie300:e0c52ded0e:dreameD9' takes more than 5000ms.
2021-11-07 19:54:44.468 [WARN ] [core.thing.internal.ThingManagerImpl] - Initializing handler for thing 'mqtt:homie300:e0c52ded0e:dreameD9' takes more than 5000ms.
2021-11-07 19:56:53.499 [WARN ] [.mqtt.homie.internal.homie300.Device] - Could not subscribe
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: No enum constant org.openhab.binding.mqtt.homie.internal.homie300.PropertyAttributes.DataTypeEnum.duration_
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1113) ~[?:?]
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2235) ~[?:?]
	at org.openhab.binding.mqtt.homie.internal.homie300.Node.subscribe(Node.java:88) ~[?:?]
	at org.openhab.binding.mqtt.homie.internal.homie300.Device.lambda$9(Device.java:278) ~[?:?]

event.log
2021-11-07 19:50:59.918 [INFO ] [openhab.event.InboxAddedEvent       ] - Discovery Result with UID 'mqtt:homie300:e0c52ded0e:dreameD9' has been added.
2021-11-07 19:51:29.936 [INFO ] [openhab.event.InboxRemovedEvent     ] - Discovery Result with UID 'mqtt:homie300:e0c52ded0e:dreameD9' has been removed.
2021-11-07 19:51:29.942 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from UNINITIALIZED to INITIALIZING
2021-11-07 19:51:29.950 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from INITIALIZING to ONLINE
2021-11-07 19:51:59.950 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Did not receive all required topics
2021-11-07 19:54:38.549 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from OFFLINE (COMMUNICATION_ERROR): Did not receive all required topics to UNINITIALIZED
2021-11-07 19:54:38.582 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from UNINITIALIZED to UNINITIALIZED (DISABLED)
2021-11-07 19:54:39.462 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from UNINITIALIZED (DISABLED) to INITIALIZING
2021-11-07 19:54:39.470 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from INITIALIZING to ONLINE
2021-11-07 19:55:09.468 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Did not receive all required topics
2021-11-07 19:56:53.290 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from OFFLINE (COMMUNICATION_ERROR): Did not receive all required topics to ONLINE
2021-11-07 19:56:53.769 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): java.lang.IllegalStateException: Attributes not yet received!
2021-11-07 20:05:09.072 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from OFFLINE (COMMUNICATION_ERROR): java.lang.IllegalStateException: Attributes not yet received! to UNINITIALIZED
2021-11-07 20:05:09.093 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from UNINITIALIZED to UNINITIALIZED (DISABLED)
2021-11-07 20:05:11.252 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from UNINITIALIZED (DISABLED) to INITIALIZING
2021-11-07 20:05:11.305 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'mqtt:homie300:e0c52ded0e:dreameD9' changed from INITIALIZING to ONLINE

**Comment by Flunkyball** [comment]
> I just readded my Valetudo Dreame D9 with 2021.11.0 with Homie enabled to openHAB 3.1.0. First I also got no channels. But a few minutes later, after the CompletionException occured in the logs, I could enable the thing and got all channels. I remember I had the same starting problems the first time I added the robot to openHAB. 

Thx for the hint. I realy never checked the channels again as the thing itself stays offline. However, the channels are really added a few minutes later and the values are up to date.

**Comment by lolodomo** [comment]
Closed by #11894
